### PR TITLE
Clang 14 and MSVC Update

### PR DESCRIPTION
**Pull request by offa**
Adds builds for the new Clang 14. Updates MSVC: 2016 has been removed and 2022 is available.

### CODE REVIEW

**Comment by rollbear** [comment]
Can you have a look @AndrewPaxie? I'm away at ACCU this week and quite busy.

**Comment by AndrewPaxie** [comment]
Yes, sure. Enjoy the conference.

File: install_libcxx.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #!/usr/bin/env bash
 
-TRUNK_VERSION="13.0.1"
+TRUNK_VERSION="14.0.1"
```
Comment (AndrewPaxie) [inline install_libcxx.sh:3]: Might want to have a closer look at this value.  By my reckoning it should be "15.0.0".

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1073,11 +1147,11 @@ jobs:
         working-directory: ${{ env.HOME }}
         run: |
           sudo ln -s /usr/include/locale.h /usr/include/xlocale.h
-          if [ "${{matrix.config.version}}" = "13" ]
+          if [ "${{matrix.config.version}}" -ge "13" ]
           then
               wget https://apt.llvm.org/llvm.sh
               chmod +x llvm.sh
-              sudo ./llvm.sh 13
+              sudo ./llvm.sh "${{matrix.config.version }}"
```
Comment (AndrewPaxie) [inline .github/workflows/ci.yml:1162]: Consider removing the space before the trailing braces.

Comment (offa) [reply .github/workflows/ci.yml]: Nice catch, thanks.

File: install_libcxx.sh (hunk 1 above)
Comment (offa) [reply install_libcxx.sh:3]: Changing to `15.0.0` fails Clang 14 builds since they try downloading from non-existing releases.

Comment (AndrewPaxie) [reply install_libcxx.sh:3]: My bad.

**Comment by offa** [comment]
Ping @rollbear @AndrewPaxie

**Comment by AndrewPaxie** [comment]
Thank you for your patience!

**Pull request merged**